core/src/main/java/hudson/model/Descriptor.java

```diff
@@ -562,7 +563,7 @@ public T newInstance(StaplerRequest req) throws FormException {
      *      Signals a problem in the submitted form.
      * @since 1.145
      */
-    public T newInstance(@CheckForNull StaplerRequest req, @Nonnull JSONObject formData) throws FormException {
+    public T newInstance(@Nullable StaplerRequest req, @Nonnull JSONObject formData) throws FormException {
```

Comment:
Maybe. But I think it's better to keep it since it highlights it to potential API users